src/query/interpret/eval.hpp

```diff
@@ -700,12 +700,14 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
       if (property_key.name == kAllPropertiesSelector.data()) {
         auto maybe_all_properties_lookup = property_value->Accept(*this);
 
-        if (maybe_all_properties_lookup.type() != TypedValue::Type::Map) {
-          throw QueryRuntimeException("Expected a map from AllPropertiesLookup, got {}.",
+        switch (maybe_all_properties_lookup.type())
+        case TypedValue::Type::Map:
+          all_properties_lookup = std::move(maybe_all_properties_lookup.ValueMap());
+        case TypedValue::Type::Null:
+          all_properties_lookup = std::move(maybe_all_properties_lookup.ValueNull());
+        default:
+          throw QueryRuntimeException("Expected a map or NULL from AllPropertiesLookup, got {}.",
```

Comment:
QueryRuntimeException is the currently message to the user. If you think this should never happen, then use LOG_FATAL to crash Memgraph. If you still believe it should be a message caused by user behaviour, then modify the message because user does not know what is AllPropertiesLookup since that's an implementation detail.

Reply:
Fixed, it’s `LOG_FATAL` now.